### PR TITLE
Implementing mouse Enter and Leave events on DiscreteColorLegend component v2

### PR DESCRIPTION
**Pull request by mcnuttandrew**
This PR is a continuation of #526 it adds tests to address the functionality suggested by @murilobast  and refactors discrete legend to be more in line with the current style. Finally it also adds the ability to specify elements in the discrete legend.

### CODE REVIEW

File: docs/markdown/legends.md [hunk 1]
```diff
@@ -18,19 +18,47 @@ Currently following types of legends are supported:
...
 
+#### onItemMouseEnter
+Type: `function`
+Default: none
+This handler is triggered either when the user's mouse enters a legend item.
+The handler passes three arguments, the corresponding item, legend index and the actual event.
+```jsx
+<DiscreteColorLegend
+...
+  onItemMouseEnter={(item, index, event)=>{
```
Comment (Apercu) [inline docs/markdown/legends.md]: missing some spaces

File: docs/markdown/legends.md [hunk 2]
```diff
@@ -18,19 +18,47 @@ Currently following types of legends are supported:
...
 Array of items that should be shown on the legend. The array should consist from either objects (`title`, optional `color` and optional `disabled` flag) or strings (treated as titles).
 
 #### onItemClick
-Type: `function(Object, number): void`  
+Type: `function(Object, number): void`
 Click callback for the item in the list. Gets the clicked item and its index as parameters.
 
+#### onItemMouseEnter
+Type: `function`
+Default: none
```
Comment (Apercu) [inline docs/markdown/legends.md]: maybe noop instead?

File: showcase/legends/horizontal-discrete-custom-palette.js [hunk 3]
```diff
@@ -42,13 +42,30 @@ const COLORS = [
...
+  render() {
+    const {hoveredItem} = this.state;
+    return (
+      <DiscreteColorLegend
+        colors={COLORS}
+        onItemMouseEnter={i => this.setState({hoveredItem: i})}
+        onItemMouseLeave={() => this.setState({hoveredItem: false})}
+        orientation="horizontal"
+        width={300}
+        items={ITEMS.map(item => {
```
Comment (Apercu) [inline showcase/legends/horizontal-discrete-custom-palette.js]:
```js
ITEMS.map(item =>
  hoveredItem === item ?
  <div>{item}<br />{'SELECTED'}</div> :
  item
)
```

also you might be missing a `key` prop too

File: src/legends/discrete-color-legend.js [hunk 4]
```diff
@@ -25,80 +25,61 @@ import PropTypes from 'prop-types';
...
+          title={item.title ? item.title : item}
+          color={item.color ? item.color : colors[i % colors.length]}
+          disabled={Boolean(item.disabled)}
           orientation={orientation}
           key={i}
-          onClick={onItemClick ?
-            () => onItemClick(initialItems[i], i) :
-            null} />
+          onClick={onItemClick ? () => onItemClick(item, i) : null}
+          onMouseEnter={onItemMouseEnter ? (e) => onItemMouseEnter(item, i, e) : null}
```
Comment (Apercu) [inline src/legends/discrete-color-legend.js]: by using noop as a default we wouldn't have to do the ternary

**Comment by Apercu** [comment]
Nice, thanks

**Comment by jchenjc** [comment]
Great work! Will this be released with a new revision (1.7.2) soon? Thanks!

**Comment by jchenjc** [comment]
Hi @mcnuttandrew, what's the best way to give this a try? Is it possible to checkout master and build a local version myself? Thanks!

**Comment by mcnuttandrew** [comment]
@jchenjc you should just be able to download 1.7.2 and use it directly now! sorry for the wait

**Comment by jchenjc** [comment]
Thank you @mcnuttandrew! :D

**Pull request merged**